envoy/listener.go

```diff
@@ -0,0 +1,45 @@
+package envoy
+
+import (
+	"github.com/Nitro/sidecar/catalog"
+)
+
+const (
+	listenerEventBufferSize = 100
+)
+
+// Listener is an internal Sidecar listener that will be hooked up when
+// config.Envoy.UseGRPCAPI is true. It only needs to know when the state
+// changes, ignoring what the actual change was.
+type Listener struct {
+	eventsChan chan catalog.ChangeEvent
+}
+
+// Chan exposes the internal events channel
+func (l *Listener) Chan() chan catalog.ChangeEvent {
+	return l.eventsChan
+}
+
+// Name returns a unique name for this listener
+func (l *Listener) Name() string {
+	// Be careful not to clash with names assigned automatically in
+	// service/service.go -> ListenerName()
+	return "internal-envoy"
+}
+
+// Managed tells Sidecar that it shouldn't try to automatically remove
+// this listener
+func (l *Listener) Managed() bool {
```

Comment:
I think the intent of this actually supposed to be the opposite, right? Sidecar is the one managing things and it should only work on things that are managed?